### PR TITLE
MSDK-531: Add OTP_Dump example for MAX32520

### PR DESCRIPTION
**Pull request by ttmut**
This PR adds OTP_Dump example to MAX32520 as requested in MSDK-531.

### CODE REVIEW

File: Examples/MAX32520/OTP_Dump/OTP_Dump.launch [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+<booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.useFileForSymbols" value="false"/>
+<booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.useProjBinaryForImage" value="true"/>
+<booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.useProjBinaryForSymbols" value="true"/>
+<booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.useRemoteTarget" value="true"/>
+<stringAttribute key="org.eclipse.cdt.dsf.gdb.DEBUG_NAME" value="arm-none-eabi-gdb"/>
+<booleanAttribute key="org.eclipse.cdt.dsf.gdb.UPDATE_THREADLIST_ON_SUSPEND" value="false"/>
+<intAttribute key="org.eclipse.cdt.launch.ATTR_BUILD_BEFORE_LAUNCH_ATTR" value="2"/>
+<stringAttribute key="org.eclipse.cdt.launch.COREFILE_PATH" value=""/>
+<stringAttribute key="org.eclipse.cdt.launch.PROGRAM_NAME" value="build/I2C_SCAN.elf"/>
+<stringAttribute key="org.eclipse.cdt.launch.PROJECT_ATTR" value="I2C_SCAN"/>
```
Comment (ozersa) [inline Examples/MAX32520/OTP_Dump/OTP_Dump.launch:51]: Example name needs to be updated in .launch file

File: Examples/MAX32520/OTP_Dump/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## Description
+
+This example demonstrates read/write to OTP area inside device's internal memory. Write test irreversibly modifies the memory and it is disabled by default.
+
+## Required Connections
+
+-   Connect a USB cable between the PC and the CN1 (USB TO UART) connector.
+-   Select RX SEL and TX SEL on headers JP7 and JP8.
```
Comment (ozersa) [inline Examples/MAX32520/OTP_Dump/README.md:8]: There is MAX32520 FTHR board too, consider this case too.

File: Examples/MAX32520/OTP_Dump/main.c [hunk 3]
```diff
@@ -0,0 +1,163 @@
+/******************************************************************************
+ * Copyright (C) 2022 Maxim Integrated Products, Inc., All rights Reserved.
```
Comment (ozersa) [inline Examples/MAX32520/OTP_Dump/main.c:2]: New file copyright date shall be 2023

**Review by ozersa — CHANGES_REQUESTED** [review]
Comment added, 
Also as I mentioned in JIRA ticket, this example shall be added for ME10/ME14 too, MA32650/MA32665

**Comment by ttmut** [comment]
> Comment added, Also as I mentioned in JIRA ticket, this example shall be added for ME10/ME14 too, MA32650/MA32665

Their documentation is not really explicit about OTP mapping though.

File: Examples/MAX32650/OTP_Dump/project.mk [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# This file can be used for project configuration.
+# It's a sibling to the core "Makefile", which offers
+# various configuration variables that you can set here
+# if the default setup isn't suitable.
+
+# See the comments in the "Makefile" for a detailed
+# description of the default behavior and the full list of
+# available options.
+
```
Comment (ozersa) [inline Examples/MAX32650/OTP_Dump/project.mk:9]: Please add below lines too.

# If you have secure version of MCU (MAX32651), set SBT=1 to generate signed binary
# For more information on how sing process works, see
# https://www.analog.com/en/education/education-library/videos/6313214207112.html
SBT=0

File: Examples/MAX32665/OTP_Dump/project.mk [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# This file can be used for project configuration.
+# It's a sibling to the core "Makefile", which offers
+# various configuration variables that you can set here
+# if the default setup isn't suitable.
+
+# See the comments in the "Makefile" for a detailed
+# description of the default behavior and the full list of
+# available options.
+
```
Comment (ozersa) [inline Examples/MAX32665/OTP_Dump/project.mk:9]: Please add below lines too
See: https://github.com/Analog-Devices-MSDK/msdk/blob/main/Examples/MAX32665/Hello_World/project.mk

# If you have secure version of MCU (MAX32666), set SBT=1 to generate signed binary
# For more information on how sign process works, see
# https://www.analog.com/en/education/education-library/videos/6313214207112.html
SBT=0

File: Libraries/Boards/MAX32665/FTHR/examples.txt [hunk 6]
```diff
@@ -41,6 +41,7 @@ I2C_SCAN
 ICC
 LP
 MAA
+OTP_Dump
```
Comment (ozersa) [inline Libraries/Boards/MAX32665/FTHR/examples.txt:44]: There is FTHR2 board too.

**Review by ozersa — APPROVED** [review]

File: Examples/MAX32520/OTP_Dump/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+-   Open an terminal application on the PC and connect to the EV kit's console UART at 115200, 8-N-1.
+
+## Expected Output
+
+```
+***** OTP Memory Read/Write Example *****
+***** This example demonstrates how you can read/write OTP memory *****
+
+
+***** MAXIM AREA *****
```
Comment (sihyung-maxim) [inline Examples/MAX32520/OTP_Dump/README.md:25]: Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar.

File: Examples/MAX32520/OTP_Dump/main.c [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+/*
+ *  Attention if you set WITH_WRITE_TEST flag
+ *  This example will search OTP and try to find 0xffffffff data
+ *  If it could be found 4 test bytes will be written in that address.
+ */
+#define WITH_WRITE_TEST 0 // set it to test write test
+
+// OTP Address
+#define OTP_MAXIM_AREA MXC_INFO0_MEM_BASE
+#define OTP_MAXIM_AREA_SIZE 0x6000
```
Comment (sihyung-maxim) [inline Examples/MAX32520/OTP_Dump/main.c:60]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32520/OTP_Dump/main.c [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+//******************************************************************************
+int main(void)
+{
+    printf("\n\n***** OTP Memory Read/Write Example *****\n");
+    printf("***** This example demonstrates how you can read/write OTP memory *****\n");
+
+    /* Dump maxim section */
+    printf("\n\n***** MAXIM AREA *****\n");
+    dump_section(OTP_MAXIM_AREA, 1024);
```
Comment (sihyung-maxim) [inline Examples/MAX32520/OTP_Dump/main.c:149]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32650/OTP_Dump/README.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+-   Open a terminal application on the PC and connect to the EV kit's console UART at 115200, 8-N-1.
+
+## Expected Output
+
+```
+***** OTP Memory Read/Write Example *****
+***** This example demonstrates how you can read/write OTP memory *****
+
+
+***** MAXIM AREA *****
```
Comment (sihyung-maxim) [inline Examples/MAX32650/OTP_Dump/README.md:25]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32650/OTP_Dump/main.c [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+/*
+ *  Attention if you set WITH_WRITE_TEST flag
+ *  This example will search OTP and try to find 0xffffffff data
+ *  If it could be found 4 test bytes will be written in that address.
+ */
+#define WITH_WRITE_TEST 0 // set it to test write test
+
+// OTP Address
+#define OTP_MAXIM_AREA MXC_INFO_MEM_BASE
+#define OTP_MAXIM_AREA_SIZE 0x1000
```
Comment (sihyung-maxim) [inline Examples/MAX32650/OTP_Dump/main.c:61]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32650/OTP_Dump/main.c [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+
+//******************************************************************************
+int main(void)
+{
+    printf("\n\n***** OTP Memory Read/Write Example *****\n");
+    printf("***** This example demonstrates how you can read/write OTP memory *****\n");
+
+    /* Dump maxim section */
+    printf("\n\n***** MAXIM AREA *****\n");
+    dump_section(OTP_MAXIM_AREA, 1024);
```
Comment (sihyung-maxim) [inline Examples/MAX32650/OTP_Dump/main.c:150]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32665/OTP_Dump/main.c [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+/*
+ *  Attention if you set WITH_WRITE_TEST flag
+ *  This example will search OTP and try to find 0xffffffff data
+ *  If it could be found 4 test bytes will be written in that address.
+ */
+#define WITH_WRITE_TEST 0 // set it to test write test
+
+// OTP Address
+#define OTP_MAXIM_AREA MXC_INFO0_MEM_BASE
+#define OTP_MAXIM_AREA_SIZE MXC_INFO_MEM_SIZE
```
Comment (sihyung-maxim) [inline Examples/MAX32665/OTP_Dump/main.c:61]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

File: Examples/MAX32665/OTP_Dump/main.c [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+//******************************************************************************
+int main(void)
+{
+    printf("\n\n***** OTP Memory Read/Write Example *****\n");
+    printf("***** This example demonstrates how you can read/write OTP memory *****\n");
+
+    /* Dump maxim section */
+    printf("\n\n***** MAXIM AREA *****\n");
+    dump_section(OTP_MAXIM_AREA, 1024);
```
Comment (sihyung-maxim) [inline Examples/MAX32665/OTP_Dump/main.c:150]: Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."

**Review by sihyung-maxim — APPROVED** [review]
Just a thought.

File: Examples/MAX32665/OTP_Dump/main.c (hunk 14 above)
Comment (ttmut) [reply Examples/MAX32665/OTP_Dump/main.c:150]: That is plausible. However, there are thousands of existing Maxim names in the SDK, are they going to be replaced as well?

Comment (ozersa) [reply Examples/MAX32665/OTP_Dump/main.c:150]: @ttmut yes, we are going to retire MAXIM name step by step. please just update the files that you touch by this PR. the rest of them will be done in future.
For OTP examples case "MANUFACTURER AREA" would be better.

@sihyung-maxim thanks for remind it.

Comment (sihyung-maxim) [reply Examples/MAX32665/OTP_Dump/main.c:150]: Don't worry about updating the copyright notices. Jake and I have a script to update the copyrights in the SDK files, and we'll run that once an ADI-appropriate notice is released to us.

**Review by ozersa — APPROVED** [review]

**Review by sihyung-maxim — COMMENTED** [review]
Don't forget the expected outputs in the examples' README.

**Review by ozersa — APPROVED** [review]

**Review by sihyung-maxim — APPROVED** [review]

**Comment by ozersa** [comment]
@sihyung-maxim could you merge it?

**Pull request merged**